File: JavaProjects/FactoryMethodSolucao/src/boletobancario/bancodobrasil/BancoDoBrasil.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package boletobancario.bancodobrasil;

import boletobancario.Banco;
import boletobancario.Boleto;

/**
 *
 * @author jeffe
 */
public class BancoDoBrasil extends Banco{

    @Override
    protected Boleto criarBoleto(int vencimento, double valor) {
        Boleto boleto = null;
        switch (vencimento) {
            case 10:
                boleto = new BancoDoBrasilBoleto10Dias(valor);
                break;
            case 30:
                boleto = new BancoDoBrasilBoleto30Dias(valor);
                break;
            case 60:
                boleto = new BancoDoBrasilBoleto60Dias(valor);
                break;
            default:
            {
                try {
                    throw new Exception("Vencimento indisponível");
                } catch (Exception ex) {
                    System.out.println(ex.getMessage());
                }
            }
        }
        return boleto;
    }
}

File: CsharpProjects/FactoryMethodSolucao/FactoryMethodSolucao/boletobancario/Bancodobrasil/BancoDoBrasil.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FactoryMethodSolucao.boletobancario.Bancodobrasil
{
    public class BancoDoBrasil : Banco
    {
        protected override Boleto criarBoleto(int vencimento, double valor)
        {
            Boleto boleto;
            switch (vencimento)
            {
                case 10:
                    boleto = new BancoDoBrasilBoleto10Dias(valor);
                    break;
                case 30:
                    boleto = new BancoDoBrasilBoleto30Dias(valor);
                    break;
                case 60:
                    boleto = new BancoDoBrasilBoleto60Dias(valor);
                    break;
                default:
                    throw new Exception("Vencimento indisponível");
            }
            return boleto;
        }
    }
}
